### DISCUSSION TITLE
How to return a .xml file via an api route.

### DISCUSSION THREAD

**Discussion by jmilani224** [vercel/next.js#24915] [Help]
I have about 1000 .xml files, and I'd like to serve the data from one of them via an api endpoint based on a user's search input. I put them in ```/public``` and it works locally, but the fetch request breaks in production. How should I format the file path so that it works in production?

```
import fs from 'fs'
import medList from '../../data/medList'
const parseString = require('xml2js').parseString;

const handler = async (req, res) => {
  const fileName = req.query.input
  if (!medList.includes(fileName)) {
    return
  }

  const filePath = `public/xml/${fileName}.nxml`

  fs.readFile(filePath, (err, data) => {
    if (err)
      throw err;
    parseString(data, async function (err, result) {
      try {
        await res.status(200);
        res.send(JSON.stringify(result));
      }
      catch (e) {
        await res.status(500);
        res.send(e);
      }
    });
  })



}

export default handler

```

**Comment by bobarros** [comment]
Is this right? nxml

```
  const filePath = `public/xml/${fileName}.nxml`
```

**Reply by jmilani224** [reply]
@bobarros Yeah, it's right - the files are from a public db. It's working locally, so it's more about how to structure the path in next.js so that it can find those files.

**Reply by bobarros** [reply]
I understood you could work locally. I was just curious about the XML format.
I never faced a problem like yours, that's why I didn't add any suggestions.
